Dockerfile

```diff
@@ -14,9 +12,19 @@ RUN apt-get update && apt-get install -y curl git make g++ bzip2 apt-transport-h
 
 ENV NODE_ENV=prd CI=true
 
+WORKDIR /app/src
+
 # RUN rm -rf app/bower_components && mkdir -p .npm && npm install && \
 #     ./node_modules/.bin/bower --allow-root install && ./node_modules/.bin/gulp dist
-RUN npm install && npm build
+ADD package.json .
+ADD bower.json .
+ADD .bowerrc .
```

Comment:
Comment for previous 3 lines – we already have `ADD . /app/src` here:

https://github.com/stellar/stellar-client/blob/241527b5a7497763014714030d979e0013204c56/Dockerfile#L5

Why do we do these extra steps?

Reply:
Ah nevermind, saw that you deleted that :)

Reply:
I actually wrote a whole post about this! This massively optimizes the build time in situations like what I was just doing, frequent rebuilds. By limiting what files we add, it limits what changes bust the cache for this part of the build https://blog.vcarl.com/optimizing-docker/